replications/internal/validator.go

```diff
@@ -2,16 +2,57 @@ package internal
 
 import (
 	"context"
+	"fmt"
+	"net/url"
+	"runtime"
 
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influxdb/v2"
 	ierrors "github.com/influxdata/influxdb/v2/kit/platform/errors"
 )
 
-func NewValidator() *stubValidator {
-	return &stubValidator{}
+func NewValidator() *noopWriteValidator {
+	return &noopWriteValidator{}
 }
 
-type stubValidator struct{}
+// noopWriteValidator checks if replication parameters are valid by attempting to write an empty payload
+// to the remote host using the configured information.
+type noopWriteValidator struct{}
 
-func (s stubValidator) ValidateReplication(ctx context.Context, config *ReplicationHTTPConfig) error {
-	return &ierrors.Error{Code: ierrors.ENotImplemented, Msg: "replication validation not implemented"}
+var userAgent = fmt.Sprintf(
+	"influxdb-oss/%s (%s) Sha/%s Date/%s",
+	influxdb.GetBuildInfo().Version,
+	runtime.GOOS,
+	influxdb.GetBuildInfo().Commit,
+	influxdb.GetBuildInfo().Date)
+
+func (s noopWriteValidator) ValidateReplication(ctx context.Context, config *ReplicationHTTPConfig) error {
+	u, err := url.Parse(config.RemoteURL)
+	if err != nil {
+		return &ierrors.Error{
+			Code: ierrors.EInvalid,
+			Msg:  fmt.Sprintf("host URL %q is invalid", config.RemoteURL),
+			Err:  err,
+		}
+	}
+	params := api.ConfigParams{
+		Host:             u,
+		UserAgent:        userAgent,
+		Token:            &config.RemoteToken,
+		AllowInsecureTLS: config.AllowInsecureTLS,
+	}
+	client := api.NewAPIClient(api.NewAPIConfig(params)).WriteApi
+
+	noopReq := client.PostWrite(ctx).
+		Org(config.RemoteOrgID.String()).
+		Bucket(config.RemoteBucketID.String()).
+		Body([]byte{})
```

Comment:
Have we checked this also doesn't error in cloud? Should we maybe add a cloud unit test that this works? It is and odd kind of request to attempt to write with no data, I can see us rejecting or otherwise flagging that it is weird.

Reply:
I've checked (it used to error, but I changed the behavior as part of an EAR last year). Agreed it would be good to have a regression test to make sure things don't revert.

Ideally we could instead have/use some sort of "can I?" API where you submit a resource/action and get back a yes/no.

Reply:
I think I prefer this kind of check (where we actually do the thing) vs a 'can I' API which might drift from the actual checks on the 'real' API. I just want to make sure it works.